ironic/drivers/modules/drac/raid.py

```diff
@@ -675,6 +760,15 @@ def _commit_to_controllers(node, controllers):
 
         driver_internal_info['raid_config_job_ids'].append(job_id)
 
+        if raid_controller not in driver_internal_info[
+                'raid_config_parameters']:
+            driver_internal_info['raid_config_parameters'].append(
+                raid_controller)
+
+    if driver_internal_info['raid_config_substep'] == 'completed':
+        del driver_internal_info['raid_config_substep']
+        del driver_internal_info['raid_config_parameters']
```

Comment:
I don't think we can delete raid_config_substep and raid_config_parameters here.  That should be done from the periodic task.

Reply:
Yes but we cannot call `_clear_raid_substep()` in else cause of if condition(`if 'raid_config_substep' in node.driver_internal_info:`) at line.no- 973 where it checks key 'raid_config_substep' otherwise it will give KeyError. that ` 'raid_config_substep' not found in _clear_raid_substep()`.

Reply:
Done